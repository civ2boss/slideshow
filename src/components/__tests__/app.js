import React from 'react';
import { shallow } from 'enzyme';
import App from '../App';

test('App renders without crashing', () => {
  global.fetch = jest.fn().mockImplementation(() => {
    return new Promise((res, rej) => {
      res({
        ok: true,
        Id: '123',
        json: function() {
          return {
            photos: {
              photo: [
                {
                  farm: 5,
                  id: '39027851635',
                  isfamily: 0,
                  isfriend: 0,
                  ispublic: 1,
                  owner: '70152224@N08',
                  secret: 'f8bd038f57',
                  server: '4757',
                  title: 'Liam. Day Fourty-Four.'
                },
                {
                  farm: 5,
                  id: '39215590324',
                  isfamily: 0,
                  isfriend: 0,
                  ispublic: 1,
                  owner: '70152224@N08',
                  secret: 'fa0c4bd546',
                  server: '4724',
                  title: 'Liam. Day Fourty-Three.'
                }
              ]
            }
          };
        }
      });
    });
  });
  shallow(<App />);
});
