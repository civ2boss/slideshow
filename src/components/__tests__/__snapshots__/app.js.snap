// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders without crashing 1`] = `
<div
  className="app"
>
  <form
    className="search"
    onSubmit={[Function]}
  >
    <input
      className="search-box"
      name="search"
      onChange={[Function]}
      placeholder="Searchâ€¦"
      type="search"
      value="puppy"
    />
  </form>
  <div
    className="slideshow"
  >
    <div
      className="viewer"
    >
      <div
        className="image"
      >
        <img
          alt=""
          src="https://farmundefined.staticflickr.com/undefined/undefined_undefined_c.jpg"
        />
        <div
          className="arrow arrow-left"
        >
          <a
            href=""
            onClick={[Function]}
          >
            &lt;
          </a>
        </div>
        <div
          className="arrow arrow-right"
        >
          <a
            href=""
            onClick={[Function]}
          >
            &gt;
          </a>
        </div>
      </div>
    </div>
    <div
      className="thumbnails"
    >
      <ul />
    </div>
  </div>
</div>
`;
